druid/src/event.rs

```diff
@@ -85,18 +86,12 @@ pub enum Event {
     /// Called when the mouse wheel or trackpad is scrolled.
     Wheel(MouseEvent),
     /// Called when a key is pressed.
-    ///
-    /// Note: the intent is for each physical key press to correspond to
-    /// a single `KeyDown` event. This is sometimes different than the
-    /// raw events provided by the platform. In particular, Windows sends
-    /// one or both of WM_KEYDOWN (a raw key code) and WM_CHAR (the
-    /// Unicode value), depending on the actual key.
-    KeyDown(KeyEvent),
+    KeyDown(KeyboardEvent),
```

Comment:
Do we think we will ever want to augment `KeyboardEvent`, such as to add a timestamp, or to expose the platform event? In that case it might make sense to still keep our own `KeyboardEvent` type, that will be a cheap conversion from `keyboard_types::KeyboardEvent`, and that will also make it easier for us to add convenience methods to that API?

It might also be nice to have our own `Modifiers` type, so that we could have more control over the API.

Reply:
Deep question with several facets.

Yes, we *absolutely* want to add timestamp at some point. But there are deep questions how to do that. I think we may want Event to become a product of timestamp and the sum of individual event types, rather than having timestamp become a new field in each of those variants.

Our own newtype over KeyboardEvent has some appeal, including those convenience methods, but I'm not convinced it overcomes the activation energy for making a new type. Certainly if we went the latter approach for timestamps that would be compelling. If we did a newtype, I would be open to having it deref to `keyboard_types::KeyboardEvent` (and probably an into).